samples/istio/mtls-mode/README.md

```diff
@@ -0,0 +1,154 @@
+# Istio - MTLS Mode
+
+In this scenario, we deploy Choreo Connect as the edge gateway where it applies API Management for the microservices.
+
+### Installation Prerequisites
+
+- [Kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+- [Kubernetes v1.20 or above](https://Kubernetes.io/docs/setup/) <br>
+
+    - Minimum CPU : 6vCPU
+    - Minimum Memory : 6GB
+
+- [Istio v1.11.x or above](https://istio.io/docs/setup/platform-setup/)
+
+### 1. Enable Istio Injection for the namespace
+
+```
+kubectl label namespace default istio-injection=enabled
+```
+
+### 2. Deploy the microservice
+
+- Deploy Httpbin service
+    ```
+    kubectl apply -f ../microservice/httpbin.yaml
+    ```
+
+- Create certificates for the ingress gateway
+
+    ```
+    ./create-certs.sh
+    ```
+    This generates the **server.crt** and **server.key** for the ingress gateway.
+
+- Deploy the certificates 
+
+    ```
+    kubectl create -n istio-system secret tls ing-gateway-credential --key=server.key --cert=server.crt
+    ```
+
+- Deploy the Gateway and Virtualservice artifacts 
+
+    ```
+    kubectl apply -f ../microservice/gw_vs_https.yaml
+    ```
+
+### 3. Invoke the microservice
+
+- Retrieve the ingress gateway URL
+
+    ```
+    kubectl get svc istio-ingressgateway -n istio-system
+  
+    Output:
+    NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                      AGE
+    istio-ingressgateway   LoadBalancer   10.99.243.134   localhost     15021:32643/TCP,80:30653/TCP,443:30717/TCP   145m
+    ```
+  
+    Use the EXTERNAL-IP which is localhost as the ingress gateway host and 443 as the ingress gateway port.
+
+
+- Add an /etc/host entry for the ingress gateway
+
+    ```
+    127.0.0.1 istio-ingressgateway.istio-system
+    ```
+ 
+- Invoke the microservice via ingress gateway
+
+    Any of the following commands can be used.
+    ```
+    curl -v -HHost:istio-ingressgateway.istio-system --resolve istio-ingressgateway.istio-system:443:localhost -k https://istio-ingressgateway.istio-system/httpbin/v1.0.0/get
+    
+    curl -v -HHost:istio-ingressgateway.istio-system --resolve istio-ingressgateway.istio-system:443:localhost --cacert server.crt https://istio-ingressgateway.istio-system/httpbin/v1.0.0/get
```

Comment:
```suggestion
    curl -v -H 'Host:istio-ingressgateway.istio-system' --resolve 'istio-ingressgateway.istio-system:443:localhost' -k https://istio-ingressgateway.istio-system/httpbin/v1.0.0/get
    
    curl -v -H 'Host:istio-ingressgateway.istio-system' --resolve 'istio-ingressgateway.istio-system:443:localhost' --cacert server.crt https://istio-ingressgateway.istio-system/httpbin/v1.0.0/get
```
Was thinking maybe it would be clearer this way